managed/services/telemetry/transform.go

```diff
@@ -41,7 +41,8 @@ func transformToJSON(config *Config, metrics []*pmmv1.ServerMetric_Metric) ([]*p
 		return nil, errors.Errorf("invalid metrics config")
 	}
 
-	firstMetric := config.Data[0].MetricName
+	// consider first metric is the beginning of the object window
+	firstMetric := metrics[0].Key
```

Comment:
small bug, that was found while testing

Reply:
There can be panic once there are no metrics returned

Reply:
if metrics cannot be collected, I think, we should not panic.

without this change if would with "return nil, errors.Errorf("invalid metrics sequence: no match with first metric")", which will be logged.

Reply:
@gen1us2k I see what you do mean, I will fix that potential case and write test for it

Reply:
I didn't see `if metrics == nil` somewhere above the code.